iocore/hostdb/HostDB.cc

```diff
@@ -634,19 +634,22 @@ HostDBProcessor::getby(Continuation *cont, const char *hostname, int len, sockad
   // Attempt to find the result in-line, for level 1 hits
   //
   if (!aforce_dns) {
+    MUTEX_TRY_LOCK(lock, cont->mutex, thread);
+    if (!lock.is_locked()) {
+      goto Ldelay;
+    }
     bool loop;
     do {
       loop = false; // Only loop on explicit set for retry.
       // find the partition lock
       //
       // TODO: Could we reuse the "mutex" above safely? I think so but not sure.
```

Comment:
Should delete this comment, it's clearly not safe because the lock on line 619 is scoped and has been dropped at this point.

Reply:
deleted.